compiler/plugins/target/ROCM/ROCMTarget.cpp

```diff
@@ -916,11 +912,31 @@ static void addAMDGPUUkernelBitcodeToGlobalEmbeddedDataDirectory() {
   });
 }
 
+// Iterate over default tuning spec embedded-data files, and insert them into
+// the EmbeddedDataDirectory singleton.
+static void addAMDGPUDefaultTuningSpecsToGlobalEmbeddedDataDirectory() {
+  EmbeddedDataDirectory::withGlobal([](EmbeddedDataDirectory &dir) {
+    const iree_file_toc_t *toc =
+        iree_default_tuning_specs_amdgpu_bytecode_create();
+    for (size_t i = 0, e = iree_default_tuning_specs_amdgpu_bytecode_size();
+         i != e; ++i) {
```

Comment:
Curious, what's the idea of doing this as opposed to

```c++
for (size_t i = 0; i != iree_default_tuning_specs_amdgpu_bytecode_size(); ++i) {
```

Reply:
It's to follow the llvm coding standards: https://llvm.org/docs/CodingStandards.html#don-t-evaluate-end-every-time-through-a-loop. It's a good default that doesn't require thinking or justification.